connection.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/connection/impl/LocalJmxConnectionProvider.java

```diff
@@ -77,15 +84,35 @@ public void close() throws IOException
 
     private void reconnect() throws IOException
     {
-        JMXServiceURL jmxUrl = new JMXServiceURL(String.format("service:jmx:rmi:///jndi/rmi://[%s]:%d/jmxrmi", myLocalhost, myPort));
+        JMXServiceURL jmxUrl = new JMXServiceURL(String.format(JMX_FORMAT_URL, myLocalhost, myPort));
         Map<String, Object> env = new HashMap<>();
         String[] credentials = this.credentialsSupplier.get();
         if (credentials != null)
         {
             env.put(JMXConnector.CREDENTIALS, credentials);
         }
+        Map<String, String> tls = this.tlsSupplier.get();
+        if (!tls.isEmpty())
+        {
+            for (Map.Entry<String, String> configEntry : tls.entrySet())
+            {
+                String key = configEntry.getKey();
+                String value = configEntry.getValue();
+
+                if (!value.isEmpty())
+                {
+                    System.setProperty(key, value);
+                }
+                else
+                {
+                    System.clearProperty(key);
+                }
+            }
+            env.put("com.sun.jndi.rmi.factory.socket", new SslRMIClientSocketFactory());
+        }
 
-        LOG.debug("Connecting JMX through {}", jmxUrl);
+        LOG.debug("Connecting JMX through {}, credentials: {}, tls: {}", jmxUrl, enabled(credentials != null),
```

Comment:
I suppose we should change this to true/false if we change the other place. However, here I think this place is more justified with more complex logic.

Reply:
Added booleans outside, so it now it should be quite simple